### PR TITLE
eval-pr: provide more instructions for autogame and multiple runs

### LINKED ISSUE #1012: Update scripts/eval-pr to support more than one PR
**Is your feature request related to a problem? Please describe.**
PR #1007 implemented a new script to aid the evaluation of a PR. Mostly for use by Freeciv21 developers and admins. It was suggested during the initial development to support a loop to evaluate more than one pull in a single call to the script.

There is also a codeFactor issue around this code block
```
  if ! [[ -e build_$pull_request ]]; then
    mkdir build_$pull_request
    cd build_$pull_request
    cmake --preset ASan -S ..
    cd ..
  fi
```

codeFactor wants this to run in a sub-process do loop. It should be fixed at the same time.

**Describe the solution you'd like**
Enable a change to the `-s`, `-a`, and `-c` command line options to support a comma separated list of PR numbers and fix the codeFactor issue.

**Describe alternatives you've considered**
In the mean time you can run the script multiple times in your own loop or string them together with `&&`. However that assumes all the builds will not fail.

**Additional context**
None

### LINKED ISSUE #1220: eval_pr: break autogame from ASan build to its own command line option
**Is your feature request related to a problem? Please describe.**
Right now the `./scripts/eval_pr` script that the admins use to automate the evaluation of pull requests has two compilation methods: Debug and ASan. The Debug method simply compiles, installs and run's cpack and stops. The ASan method compiles with the ASan preset and the automatically runs an autogame.

**Describe the solution you'd like**
ASan does not compile with `Debug`, but instead use `RelWithDebInfo` due to a collection of factors. When running an autogame, it is sometimes useful to run the server and client inside of a debugger (such as gdb) so if an assertion occurs or a segfault of some kind occurs a tester can get a backtrace to aid an issue later on.

What I would like is to have a separate function for calling an autogame in the script. The regular Debug and ASan presets can be used and then if you want the script to run an autogame, you can do that.

**Describe alternatives you've considered**
Could also simply remove the autostart of the autogame from ASan and simply `echo` out instructions on how to start one if a user wants to go that route.

**Additional context**
None

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1220
Closes #1012

### CODE REVIEW

**Comment by lmoureaux** [comment]
#1012 says:

> There is also a codeFactor issue around this code block
> ```sh
>  if ! [[ -e build_$pull_request ]]; then
>    mkdir build_$pull_request
>    cd build_$pull_request
>    cmake --preset ASan -S ..
>    cd ..
>  fi
> ```
> codeFactor wants this to run in a sub-process do loop. It should be fixed at the same time.

I don't think this is fixed in this PR.

**Comment by jwrober** [comment]
The MacOS failure seems unrelated. Some issue with `gettext`.

**Comment by lmoureaux** [comment]
Agreed

File: scripts/eval-pr [hunk 1]
```diff
@@ -78,19 +85,62 @@ base_setup ()
...
+    echo ""
+    echo "-- If you wish to run an autogame to test the code in PR $pull_request, then open another terminal"
+    echo "--  and run this command:"
+    echo "      $PWD/build_$pull_request/freeciv21-server -r $PWD/data/test-autogame.serv -t"
+    echo ""
+    echo "-- If you want to observe the autogame, open another terminal and run this command to open a client"
+    echo "      $PWD/build_$pull_request/freeciv21-client -a -p 5556 -s localhost"
+    echo ""
+    echo "-- Lastly, if you wish to run the autogame server or observing client through a debugger, then"
+    echo "    prefix the previous two command lines with \"gdb --args\"."
```
Comment (lmoureaux) [inline scripts/eval-pr:105]: This is inconsistent with other long messages, usually the `--` is reprinted

File: scripts/eval-pr [hunk 2]
```diff
@@ -78,19 +85,62 @@ base_setup ()
...
+    echo "      export ASAN_OPTIONS=\"halt_on_error=0\" && $PWD/build_$pull_request/freeciv21-server -r $PWD/data/test-autogame.serv -t"
+    echo ""
+    echo "-- If you want to observe the autogame, open another terminal and run this command to open a client"
+    echo ""
+    echo "      $PWD/build_$pull_request/freeciv21-client -a -p 5556 -s localhost"
+    echo ""
+    echo "-- Lastly, The ASan preset compiles with the \"RelWithDebInfo\" configuration instead of the"
+    echo "    \"Debug\" configuration. Running the code through a debugger, such as gdb, can have unintended"
+    echo "    effects. If you wish to run the autogame server or observing client through a debugger, then"
+    echo "    prefix the previous two command lines with \"gdb --args\"."
```
Comment (lmoureaux) [inline scripts/eval-pr:123]: Same as above, `--` for long messages

File: scripts/eval-pr [hunk 3]
```diff
@@ -78,19 +85,62 @@ base_setup ()
...
+{
+  if [[ "$asan" == "false" ]]; then
+    echo "-- You can open the client with this command in another terminal to test PR $pull_request:"
+    echo "      $PWD/build_$pull_request/freeciv21-client"
+    echo ""
+    echo "-- If you wish to run the client through a debugger, such as gdb, then prefix \"gdb\" in front of"
+    echo "--  the command above."
+    echo ""
+    echo "-- If you wish to run an autogame to test the code in PR $pull_request, then open another terminal"
+    echo "--  and run this command:"
```
Comment (lmoureaux) [inline scripts/eval-pr:98]: Add an empty line here

File: scripts/eval-pr [hunk 4]
```diff
@@ -78,19 +85,62 @@ base_setup ()
   fi
 }
 
-start ()
+close_message ()
+{
+  if [[ "$asan" == "false" ]]; then
+    echo "-- You can open the client with this command in another terminal to test PR $pull_request:"
+    echo "      $PWD/build_$pull_request/freeciv21-client"
+    echo ""
```
Comment (lmoureaux) [inline scripts/eval-pr:93]:
```suggestion
    echo
```
(and in many other places, empty `echo` prints a newline)

**Review by lmoureaux — COMMENTED** [review]
Running the ASAN build after a normal build, I get the following:
```
-- Applying the .diff via patch.

patching file server/unithand.cpp
Reversed (or previously applied) patch detected!  Assume -R? [n] 
```

**Pull request merged**